[oauth] Error fetching access token. Invalid oauth code? Please generate a new one.

I am running on a Windows 10 PC:

Operating system: Windows 10/10.0 (amd64)
Java Runtime: Azul Systems, Inc. 17.0.6 (Zulu17.40+19-CA)

Since upgrading from 3.4.2 to 4.0.0-SNAPSHOT and now 4.0.0.M1 I am getting errors trying to authenticate for Google TTS:

```Error fetching access token. Invalid oauth code? Please generate a new one.```

When logged using TRACE logging:
```
log:set TRACE org.openhab.voice.googletts
log:set TRACE org.openhab.core.auth.oauth2client
```
I get:
```
11:22:33.943 [DEBUG] [e.googletts.internal.GoogleTTSService] - bundle org.openhab.voice.googletts:4.0.0.M1 (257)[org.openhab.voice.googletts.internal.GoogleTTSService(336)] : ConfigurableComponentHolder configuration updated for pid org.openhab.voice.googletts with change count 61
11:22:33.944 [DEBUG] [e.googletts.internal.GoogleTTSService] - bundle org.openhab.voice.googletts:4.0.0.M1 (257)[org.openhab.voice.googletts.internal.GoogleTTSService(336)] : Querying state active
11:22:33.945 [DEBUG] [e.googletts.internal.GoogleTTSService] - bundle org.openhab.voice.googletts:4.0.0.M1 (257)[org.openhab.voice.googletts.internal.GoogleTTSService(336)] : Querying state active
11:22:33.945 [DEBUG] [e.googletts.internal.GoogleTTSService] - bundle org.openhab.voice.googletts:4.0.0.M1 (257)[org.openhab.voice.googletts.internal.GoogleTTSService(336)] : invoking modified: updateConfig: parameters [org.apache.felix.scr.impl.helper.ReadOnlyDictionary]
11:22:33.946 [DEBUG] [e.googletts.internal.GoogleTTSService] - Updating configuration
11:22:33.946 [TRACE] [e.googletts.internal.GoogleTTSService] - New configuration: GoogleTTSConfig{pitch=0.0, speakingRate=1.0, volumeGainDb=0.0, purgeCache=true}
11:22:33.947 [DEBUG] [ice.googletts.internal.GoogleCloudAPI] - Trying to get access and refresh tokens.
11:22:34.122 [DEBUG] [.oauth2client.internal.OAuthConnector] - Oauth request parameter grant_type, value authorization_code
11:22:34.123 [DEBUG] [.oauth2client.internal.OAuthConnector] - Oauth request parameter code, value <REDACT>
11:22:34.124 [DEBUG] [.oauth2client.internal.OAuthConnector] - Oauth request parameter redirect_uri, value https://www.google.com
11:22:34.124 [DEBUG] [.oauth2client.internal.OAuthConnector] - Setting authentication for clientId <REDACT> Using basic auth false
11:22:35.564 [DEBUG] [ice.googletts.internal.GoogleCloudAPI] - Error fetching access token: Exception in oauth communication, grant type authorization_code
org.openhab.core.auth.client.oauth2.OAuthException: Exception in oauth communication, grant type authorization_code
        at org.openhab.core.auth.oauth2client.internal.OAuthConnector.doRequest(OAuthConnector.java:331) ~[?:?]
        at org.openhab.core.auth.oauth2client.internal.OAuthConnector.grantTypeAuthorizationCode(OAuthConnector.java:220) ~[?:?]
        at org.openhab.core.auth.oauth2client.internal.OAuthClientServiceImpl.getAccessTokenResponseByAuthorizationCode(OAuthClientServiceImpl.java:211) ~[?:?]
        at org.openhab.voice.googletts.internal.GoogleCloudAPI.getAccessToken(GoogleCloudAPI.java:192) ~[?:?]
        at org.openhab.voice.googletts.internal.GoogleCloudAPI.setConfig(GoogleCloudAPI.java:154) ~[?:?]
        at org.openhab.voice.googletts.internal.GoogleTTSService.updateConfig(GoogleTTSService.java:240) ~[?:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
        at org.apache.felix.scr.impl.inject.methods.BaseMethod.invokeMethod(BaseMethod.java:245) ~[?:?]
        at org.apache.felix.scr.impl.inject.methods.BaseMethod.access$500(BaseMethod.java:41) ~[?:?]
        at org.apache.felix.scr.impl.inject.methods.BaseMethod$Resolved.invoke(BaseMethod.java:687) ~[?:?]
        at org.apache.felix.scr.impl.inject.methods.BaseMethod.invoke(BaseMethod.java:531) ~[?:?]
        at org.apache.felix.scr.impl.inject.methods.ActivateMethod.invoke(ActivateMethod.java:317) ~[?:?]
        at org.apache.felix.scr.impl.inject.methods.ActivateMethod.invoke(ActivateMethod.java:307) ~[?:?]
        at org.apache.felix.scr.impl.manager.SingleComponentManager.invokeModifiedMethod(SingleComponentManager.java:856) ~[?:?]
        at org.apache.felix.scr.impl.manager.SingleComponentManager.modify(SingleComponentManager.java:810) ~[?:?]
        at org.apache.felix.scr.impl.manager.SingleComponentManager.reconfigure(SingleComponentManager.java:725) ~[?:?]
        at org.apache.felix.scr.impl.manager.SingleComponentManager.reconfigure(SingleComponentManager.java:686) ~[?:?]
        at org.apache.felix.scr.impl.manager.ConfigurableComponentHolder.configurationUpdated(ConfigurableComponentHolder.java:440) ~[?:?]
        at org.apache.felix.scr.impl.manager.RegionConfigurationSupport.configurationEvent(RegionConfigurationSupport.java:347) ~[?:?]
        at org.apache.felix.scr.impl.manager.RegionConfigurationSupport$2.configurationEvent(RegionConfigurationSupport.java:115) ~[?:?]
        at org.apache.felix.cm.impl.ConfigurationManager$FireConfigurationEvent.sendEvent(ConfigurationManager.java:1723) ~[?:?]
        at org.apache.felix.cm.impl.ConfigurationManager$FireConfigurationEvent.run(ConfigurationManager.java:1664) ~[?:?]
        at org.apache.felix.cm.impl.UpdateThread.run0(UpdateThread.java:122) ~[?:?]
        at org.apache.felix.cm.impl.UpdateThread.run(UpdateThread.java:84) ~[?:?]
        at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: com.google.gson.JsonIOException: Failed making field 'java.time.Instant#seconds' accessible; either change its visibility or write a custom TypeAdapter for its declaring type
        at com.google.gson.internal.reflect.ReflectionHelper.makeAccessible(ReflectionHelper.java:23) ~[?:?]
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.getBoundFields(ReflectiveTypeAdapterFactory.java:203) ~[?:?]
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.create(ReflectiveTypeAdapterFactory.java:112) ~[?:?]
        at com.google.gson.Gson.getAdapter(Gson.java:531) ~[?:?]
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.createBoundField(ReflectiveTypeAdapterFactory.java:137) ~[?:?]
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.getBoundFields(ReflectiveTypeAdapterFactory.java:211) ~[?:?]
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.create(ReflectiveTypeAdapterFactory.java:112) ~[?:?]
        at com.google.gson.Gson.getAdapter(Gson.java:531) ~[?:?]
        at com.google.gson.Gson.fromJson(Gson.java:1057) ~[?:?]
        at com.google.gson.Gson.fromJson(Gson.java:1016) ~[?:?]
        at com.google.gson.Gson.fromJson(Gson.java:959) ~[?:?]
        at com.google.gson.Gson.fromJson(Gson.java:927) ~[?:?]
        at org.openhab.core.auth.oauth2client.internal.OAuthConnector.doRequest(OAuthConnector.java:308) ~[?:?]
        ... 27 more
Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make field private final long java.time.Instant.seconds accessible: module java.base does not "opens java.time" to unnamed module @72d10ef7
        at java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:354) ~[?:?]
        at java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297) ~[?:?]
        at java.lang.reflect.Field.checkCanSetAccessible(Field.java:178) ~[?:?]
        at java.lang.reflect.Field.setAccessible(Field.java:172) ~[?:?]
        at com.google.gson.internal.reflect.ReflectionHelper.makeAccessible(ReflectionHelper.java:20) ~[?:?]
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.getBoundFields(ReflectiveTypeAdapterFactory.java:203) ~[?:?]
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.create(ReflectiveTypeAdapterFactory.java:112) ~[?:?]
        at com.google.gson.Gson.getAdapter(Gson.java:531) ~[?:?]
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.createBoundField(ReflectiveTypeAdapterFactory.java:137) ~[?:?]
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.getBoundFields(ReflectiveTypeAdapterFactory.java:211) ~[?:?]
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.create(ReflectiveTypeAdapterFactory.java:112) ~[?:?]
        at com.google.gson.Gson.getAdapter(Gson.java:531) ~[?:?]
        at com.google.gson.Gson.fromJson(Gson.java:1057) ~[?:?]
        at com.google.gson.Gson.fromJson(Gson.java:1016) ~[?:?]
        at com.google.gson.Gson.fromJson(Gson.java:959) ~[?:?]
        at com.google.gson.Gson.fromJson(Gson.java:927) ~[?:?]
        at org.openhab.core.auth.oauth2client.internal.OAuthConnector.doRequest(OAuthConnector.java:308) ~[?:?]
        ... 27 more
11:22:35.574 [WARN ] [ice.googletts.internal.GoogleCloudAPI] - Error initializing Google Cloud TTS service: Error fetching access token. Invalid authcode? Please generate a new one.
11:22:35.575 [DEBUG] [ice.googletts.internal.GoogleCloudAPI] - Cache purged.
11:22:35.576 [DEBUG] [e.googletts.internal.GoogleTTSService] - bundle org.openhab.voice.googletts:4.0.0.M1 (257)[org.openhab.voice.googletts.internal.GoogleTTSService(336)] : invoked modified: updateConfig
11:22:35.577 [DEBUG] [e.googletts.internal.GoogleTTSService] - bundle org.openhab.voice.googletts:4.0.0.M1 (257)[org.openhab.voice.googletts.internal.GoogleTTSService(336)] : No change in target property for dependency $000: currently registered: true
11:22:35.577 [DEBUG] [e.googletts.internal.GoogleTTSService] - bundle org.openhab.voice.googletts:4.0.0.M1 (257)[org.openhab.voice.googletts.internal.GoogleTTSService(336)] : No change in target property for dependency $001: currently registered: true
11:22:35.578 [DEBUG] [e.googletts.internal.GoogleTTSService] - bundle org.openhab.voice.googletts:4.0.0.M1 (257)[org.openhab.voice.googletts.internal.GoogleTTSService(336)] : No change in target property for dependency osgi.ds.satisfying.condition: currently registered: true
11:22:35.578 [DEBUG] [e.googletts.internal.GoogleTTSService] - bundle org.openhab.voice.googletts:4.0.0.M1 (257)[org.openhab.voice.googletts.internal.GoogleTTSService(336)] : Querying state active
11:22:35.580 [DEBUG] [e.googletts.internal.GoogleTTSService] - bundle org.openhab.voice.googletts:4.0.0.M1 (257)[org.openhab.voice.googletts.internal.GoogleTTSService(336)] : ImmediateComponentHolder Finished configuring the dependency managers for component for pid org.openhab.voice.googletts
11:22:35.581 [DEBUG] [e.googletts.internal.GoogleTTSService] - bundle org.openhab.voice.googletts:4.0.0.M1 (257)[org.openhab.voice.googletts.internal.GoogleTTSService(336)] : ImmediateComponentHolder Will not enable component for pid org.openhab.voice.googletts: holder enabled state: true, metadata enabled: true
```
I have retested with 3.4.2 (Java 11) and with a CLEAN 4.0.0-SNAPSHOT (only Google TTS installed) and both work fine.

Any assistance would be appreciated.

Please note, the call stack reported in the [forum post](https://community.openhab.org/t/oh4-0-0-snapshot-and-m1-google-tts-exception/145046/11) is not the same as in the PR description:
```
[ERROR] [e.googlestt.internal.GoogleSTTService] - bundle org.openhab.voice.googlestt:4.0.0.M1 (258)[org.openhab.voice.googlestt.internal.GoogleSTTService(338)] : The modified method has thrown an exception
com.google.gson.JsonIOException: Failed making field 'java.time.LocalDateTime#date' accessible; either change its visibility or write a custom TypeAdapter for its declaring type
        at com.google.gson.internal.reflect.ReflectionHelper.makeAccessible(ReflectionHelper.java:23) ~[?:?]
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.getBoundFields(ReflectiveTypeAdapterFactory.java:203) ~[?:?]
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.create(ReflectiveTypeAdapterFactory.java:112) ~[?:?]
        at com.google.gson.Gson.getAdapter(Gson.java:531) ~[?:?]
        at com.google.gson.Gson.toJson(Gson.java:778) ~[?:?]
        at com.google.gson.Gson.toJson(Gson.java:756) ~[?:?]
        at com.google.gson.Gson.toJson(Gson.java:711) ~[?:?]
        at com.google.gson.Gson.toJson(Gson.java:691) ~[?:?]
        at org.openhab.core.auth.oauth2client.internal.OAuthStoreHandlerImpl$StorageFacade.put(OAuthStoreHandlerImpl.java:366) ~[?:?]
        at org.openhab.core.auth.oauth2client.internal.OAuthStoreHandlerImpl.savePersistedParams(OAuthStoreHandlerImpl.java:164) ~[?:?]
        at org.openhab.core.auth.oauth2client.internal.OAuthClientServiceImpl.createInstance(OAuthClientServiceImpl.java:131) ~[?:?]
        at org.openhab.core.auth.oauth2client.internal.OAuthFactoryImpl.createOAuthClientService(OAuthFactoryImpl.java:83) ~[?:?]
        at org.openhab.voice.googlestt.internal.GoogleSTTService.updateConfig(GoogleSTTService.java:163) ~[?:?]
        at org.openhab.voice.googlestt.internal.GoogleSTTService.modified(GoogleSTTService.java:117) ~[?:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
        at org.apache.felix.scr.impl.inject.methods.BaseMethod.invokeMethod(BaseMethod.java:245) ~[?:?]
        at org.apache.felix.scr.impl.inject.methods.BaseMethod.access$500(BaseMethod.java:41) ~[?:?]
        at org.apache.felix.scr.impl.inject.methods.BaseMethod$Resolved.invoke(BaseMethod.java:687) ~[?:?]
        at org.apache.felix.scr.impl.inject.methods.BaseMethod.invoke(BaseMethod.java:531) ~[?:?]
        at org.apache.felix.scr.impl.inject.methods.ActivateMethod.invoke(ActivateMethod.java:317) ~[?:?]
        at org.apache.felix.scr.impl.inject.methods.ActivateMethod.invoke(ActivateMethod.java:307) ~[?:?]
        at org.apache.felix.scr.impl.manager.SingleComponentManager.invokeModifiedMethod(SingleComponentManager.java:856) ~[?:?]
        at org.apache.felix.scr.impl.manager.SingleComponentManager.modify(SingleComponentManager.java:810) ~[?:?]
        at org.apache.felix.scr.impl.manager.SingleComponentManager.reconfigure(SingleComponentManager.java:725) ~[?:?]
        at org.apache.felix.scr.impl.manager.SingleComponentManager.reconfigure(SingleComponentManager.java:686) ~[?:?]
        at org.apache.felix.scr.impl.manager.ConfigurableComponentHolder.configurationUpdated(ConfigurableComponentHolder.java:440) ~[?:?]
        at org.apache.felix.scr.impl.manager.RegionConfigurationSupport.configurationEvent(RegionConfigurationSupport.java:347) ~[?:?]
        at org.apache.felix.scr.impl.manager.RegionConfigurationSupport$2.configurationEvent(RegionConfigurationSupport.java:115) ~[?:?]
        at org.apache.felix.cm.impl.ConfigurationManager$FireConfigurationEvent.sendEvent(ConfigurationManager.java:1723) ~[?:?]
        at org.apache.felix.cm.impl.ConfigurationManager$FireConfigurationEvent.run(ConfigurationManager.java:1664) ~[?:?]
        at org.apache.felix.cm.impl.UpdateThread.run0(UpdateThread.java:122) ~[?:?]
        at org.apache.felix.cm.impl.UpdateThread.run(UpdateThread.java:84) ~[?:?]
        at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make field private final java.time.LocalDate java.time.LocalDateTime.date accessible: module java.base does not "opens java.time" to unnamed module @72d10ef7
        at java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:354) ~[?:?]
        at java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297) ~[?:?]
        at java.lang.reflect.Field.checkCanSetAccessible(Field.java:178) ~[?:?]
        at java.lang.reflect.Field.setAccessible(Field.java:172) ~[?:?]
        at com.google.gson.internal.reflect.ReflectionHelper.makeAccessible(ReflectionHelper.java:20) ~[?:?]
        ... 35 more
```

--- a/bundles/org.openhab.core.auth.oauth2client/src/main/java/org/openhab/core/auth/oauth2client/internal/OAuthConnector.java
+++ b/bundles/org.openhab.core.auth.oauth2client/src/main/java/org/openhab/core/auth/oauth2client/internal/OAuthConnector.java
@@ -19,6 +19,9 @@ import java.lang.reflect.InvocationTargetException;
 import java.net.URLEncoder;
 import java.nio.charset.StandardCharsets;
 import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeParseException;
 import java.util.Base64;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.TimeoutException;
@@ -43,6 +46,7 @@ import org.slf4j.LoggerFactory;
 import com.google.gson.FieldNamingPolicy;
 import com.google.gson.Gson;
 import com.google.gson.GsonBuilder;
+import com.google.gson.JsonDeserializer;
 import com.google.gson.JsonSyntaxException;
 
 /**
@@ -65,7 +69,15 @@ public class OAuthConnector {
 
     public OAuthConnector(HttpClientFactory httpClientFactory, @Nullable String deserializerClassName) {
         this.httpClientFactory = httpClientFactory;
-        GsonBuilder gsonBuilder = new GsonBuilder().setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES);
+        GsonBuilder gsonBuilder = new GsonBuilder().setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES)
+                .registerTypeAdapter(Instant.class, (JsonDeserializer<Instant>) (json, typeOfT, context) -> {
+                    try {
+                        return Instant.parse(json.getAsString());
+                    } catch (DateTimeParseException e) {
+                        return LocalDateTime.parse(json.getAsString()).atZone(ZoneId.systemDefault()).toInstant();
+                    }
+                });
+
         if (deserializerClassName != null) {
             try {
                 Class<?> deserializerClass = Class.forName(deserializerClassName);

--- a/bundles/org.openhab.core.auth.oauth2client/src/main/java/org/openhab/core/auth/oauth2client/internal/OAuthStoreHandlerImpl.java
+++ b/bundles/org.openhab.core.auth.oauth2client/src/main/java/org/openhab/core/auth/oauth2client/internal/OAuthStoreHandlerImpl.java
@@ -19,6 +19,7 @@ import java.time.Instant;
 import java.time.LocalDateTime;
 import java.time.ZoneId;
 import java.time.format.DateTimeParseException;
+import java.time.temporal.ChronoUnit;
 import java.util.HashSet;
 import java.util.LinkedHashSet;
 import java.util.Map;
@@ -221,12 +222,12 @@ public class OAuthStoreHandlerImpl implements OAuthStoreHandler {
         }
     }
 
-    private boolean isExpired(@Nullable LocalDateTime lastUsed) {
+    private boolean isExpired(@Nullable Instant lastUsed) {
         if (lastUsed == null) {
             return false;
         }
         // (last used + 183 days < now) then it is expired
-        return lastUsed.plusDays(EXPIRE_DAYS).isBefore(LocalDateTime.now());
+        return lastUsed.plus(EXPIRE_DAYS, ChronoUnit.DAYS).isBefore(Instant.now());
     }
 
     /**
@@ -306,28 +307,14 @@ public class OAuthStoreHandlerImpl implements OAuthStoreHandler {
                     }
                 } else if (LAST_USED.equals(recordType)) {
                     try {
-                        LocalDateTime lastUsedDate = gson.fromJson(value, LocalDateTime.class);
+                        Instant lastUsedDate = gson.fromJson(value, Instant.class);
                         return lastUsedDate;
                     } catch (Exception e) {
                         logger.info("Unable to deserialize json, reset LAST_USED to now.  json:\n{}", value);
-                        return LocalDateTime.now();
+                        return Instant.now();
                     }
                 }
                 return null;
-            } finally {
-                storageLock.unlock();
-            }
-        }
-
-        public void put(String handle, @Nullable LocalDateTime lastUsed) {
-            storageLock.lock();
-            try {
-                if (lastUsed == null) {
-                    storage.put(LAST_USED.getKey(handle), (String) null);
-                } else {
-                    String gsonStr = gson.toJson(lastUsed);
-                    storage.put(LAST_USED.getKey(handle), gsonStr);
-                }
             } finally {
                 storageLock.unlock();
             }
@@ -341,7 +328,7 @@ public class OAuthStoreHandlerImpl implements OAuthStoreHandler {
                 } else {
                     String gsonAccessTokenStr = gson.toJson(accessTokenResponse);
                     storage.put(ACCESS_TOKEN_RESPONSE.getKey(handle), gsonAccessTokenStr);
-                    String gsonDateStr = gson.toJson(LocalDateTime.now());
+                    String gsonDateStr = gson.toJson(Instant.now());
                     storage.put(LAST_USED.getKey(handle), gsonDateStr);
 
                     if (!allHandles.contains(handle)) {
@@ -363,7 +350,7 @@ public class OAuthStoreHandlerImpl implements OAuthStoreHandler {
                 } else {
                     String gsonPersistedParamsStr = gson.toJson(persistedParams);
                     storage.put(SERVICE_CONFIGURATION.getKey(handle), gsonPersistedParamsStr);
-                    String gsonDateStr = gson.toJson(LocalDateTime.now());
+                    String gsonDateStr = gson.toJson(Instant.now());
                     storage.put(LAST_USED.getKey(handle), gsonDateStr);
                     if (!allHandles.contains(handle)) {
                         // update all handles index
@@ -412,7 +399,7 @@ public class OAuthStoreHandlerImpl implements OAuthStoreHandler {
                     if (handlesSSV != null) {
                         String[] handles = handlesSSV.trim().split(" ");
                         for (String handle : handles) {
-                            LocalDateTime lastUsed = (LocalDateTime) get(handle, LAST_USED);
+                            Instant lastUsed = (Instant) get(handle, LAST_USED);
                             if (isExpired(lastUsed)) {
                                 removeByHandle(handle);
                             }

--- a/bundles/org.openhab.core.auth.oauth2client/src/test/java/org/openhab/core/auth/oauth2client/internal/OAuthStoreHandlerTest.java
+++ b/bundles/org.openhab.core.auth.oauth2client/src/test/java/org/openhab/core/auth/oauth2client/internal/OAuthStoreHandlerTest.java
@@ -10,7 +10,7 @@
  *
  * SPDX-License-Identifier: EPL-2.0
  */
-package org.openhab.core.auth.oauth2client;
+package org.openhab.core.auth.oauth2client.internal;
 
 import static org.hamcrest.CoreMatchers.notNullValue;
 import static org.hamcrest.MatcherAssert.assertThat;
@@ -32,8 +32,6 @@ import org.mockito.junit.jupiter.MockitoExtension;
 import org.mockito.junit.jupiter.MockitoSettings;
 import org.mockito.quality.Strictness;
 import org.openhab.core.auth.client.oauth2.AccessTokenResponse;
-import org.openhab.core.auth.oauth2client.internal.OAuthStoreHandlerImpl;
-import org.openhab.core.auth.oauth2client.internal.StorageRecordType;
 import org.openhab.core.storage.Storage;
 import org.openhab.core.storage.StorageService;
 import org.openhab.core.test.storage.VolatileStorage;
@@ -55,14 +53,14 @@ public class OAuthStoreHandlerTest {
     private @NonNullByDefault({}) OAuthStoreHandlerImpl storeHandler;
 
     @BeforeEach
-    public void initialize() throws IOException {
+    void initialize() throws IOException {
         storage = new VolatileStorage<>();
         Mockito.doReturn(storage).when(storageService).getStorage(STORE_NAME);
         storeHandler = new OAuthStoreHandlerImpl(storageService);
     }
 
     @Test
-    public void loadAccessTokenResponseWhenCreatedOnIsLocalDateTime() throws GeneralSecurityException {
+    void loadAccessTokenResponseWhenCreatedOnIsLocalDateTime() throws GeneralSecurityException {
         final String handle = "test";
         final String createdOn = "2022-08-14T21:21:05.568991";
         final Instant expected = LocalDateTime.parse(createdOn).atZone(ZoneId.systemDefault()).toInstant();
@@ -77,7 +75,7 @@ public class OAuthStoreHandlerTest {
     }
 
     @Test
-    public void loadAccessTokenResponseWhenCreatedOnIsInstant() throws GeneralSecurityException {
+    void loadAccessTokenResponseWhenCreatedOnIsInstant() throws GeneralSecurityException {
         final String handle = "test";
         final String createdOn = "2022-08-14T19:21:05.568991Z";
         final Instant expected = Instant.parse(createdOn);
@@ -89,6 +87,13 @@ public class OAuthStoreHandlerTest {
         if (response != null) {
             assertThat(response.getCreatedOn(), is(expected));
         }
+    }
+
+    @Test
+    void savePersistedParamsShouldNotThrow() {
+        final String handle = "test";
+
+        storeHandler.savePersistedParams(handle, new PersistedParams());
     }
 
     private String getJsonforCreatedOn(String createdOn) {

